stdlib/Base64/src/decode.jl

```diff
@@ -32,17 +32,19 @@ julia> String(read(iob64_decode))
 "Hello!"
 ```
 """
-struct Base64DecodePipe <: IO
-    io::IO
+struct Base64DecodePipe{T <: IO} <: IO
+    io::T
     buffer::Buffer
     rest::Vector{UInt8}
 
-    function Base64DecodePipe(io::IO)
+    function Base64DecodePipe{T}(io::T) where {T <: IO}
         buffer = Buffer(512)
-        return new(io, buffer, UInt8[])
+        return new{T}(io, buffer, UInt8[])
     end
 end
 
+Base64DecodePipe(io::IO) = Base64DecodePipe{typeof(io)}(io)
```

Comment:
```suggestion
Base64DecodePipe(io::IO) = Base64DecodePipe{IO}(io)
```

Doesn't seem like this PR makes things faster, so by default it seems better to avoid making the compile slower and more complex?